src/hotspot/share/opto/loopPredicate.cpp

```diff
@@ -99,40 +99,38 @@ void PhaseIdealLoop::register_control(Node* n, IdealLoopTree *loop, Node* pred,
 // We will create a region to guard the uct call if there is no one there.
 // The continuation projection (if_cont) of the new_iff is returned which
 // is an IfTrue projection. This code is also used to clone predicates to cloned loops.
-IfProjNode* PhaseIdealLoop::create_new_if_for_predicate(ParsePredicateSuccessProj* parse_predicate_proj, Node* new_entry,
-                                                        Deoptimization::DeoptReason reason,
+IfTrueNode* PhaseIdealLoop::create_new_if_for_predicate(ParsePredicateSuccessProj* parse_predicate_success_proj,
+                                                        Node* new_entry, const Deoptimization::DeoptReason reason,
                                                         const int opcode, const bool rewire_uncommon_proj_phi_inputs) {
-  assert(parse_predicate_proj->is_uncommon_trap_if_pattern(reason), "must be a uct if pattern!");
-  ParsePredicateNode* parse_predicate = parse_predicate_proj->in(0)->as_ParsePredicate();
-
-  ProjNode* uncommon_proj = parse_predicate->proj_out(false);
-  Node* uct_region = uncommon_proj->unique_ctrl_out();
-  assert(uct_region->is_Region() || uct_region->is_Call(), "must be a region or call uct");
```

Comment:
Did you want to remove this assert? Or is it elsewhere now?

Reply:
It's already covered in `ParsePredicate::uncommon_trap()`:
https://github.com/openjdk/jdk/blob/3e185c70feef3febf75c58a5d4d394a4b772105f/src/hotspot/share/opto/ifnode.cpp#L2149-L2154